src/main/groovy/com/cloudcard/photoDownloader/TouchNetClient.groovy

```diff
@@ -58,6 +63,72 @@ class TouchNetClient {
         log.info("        TouchNet Origin ID : $originId")
     }
 
+    @Override
+    String getSystemName() {
+        return "TouchNet"
+    }
+
+    @Override
+    void putPhoto(String accountId, String photoBase64) {
+        ensureSession()
+
+        Map request = [
+                SessionID: sessionId,
+                AccountID: accountId,
+                PhotoBase64: photoBase64,
+                ForcePrintedFlag: false
+        ]
+
+        TouchNetResponse response = doApiRequest("Account Photo Approve", "account/photo/approve", request)
+        if (!response.success) {
+            String message = response.getErrorMessage() ?: "Unknown TouchNet error"
+
+            if (isRateLimitError(message)) {
+                for (int i = 1; i <= MAX_TRIES; i++) {
+                    RETRY_DELAY_MS * i
+                    log.warn("Rate limit hit, retrying in ${RETRY_DELAY_MS}ms (attempt $i/$MAX_TRIES)")
+                    Thread.sleep(RETRY_DELAY_MS)
+                    response = doApiRequest("Account Photo Approve", "account/photo/approve", request)
+                    if (response.success) return
```

Comment:
I think we need slightly more robust wrapping logic for the retry, and some more robust objects getting returned by this client.